gateway-service/src/main/java/org/zowe/apiml/gateway/conformance/VerificationOnboardService.java

```diff
@@ -74,58 +70,114 @@ public Optional<String> findSwaggerUrl(Map<String, String> metadata) {
-                result.add("Calling endpoint at " + endpoint.getUrl() + " gives undocumented " + response.getStatusCode().value()
-                    + " status code, documented responses are:" + endpoint.getValidResponses().get("GET"));
-            }
+        ResponseEntity<String> responseNoSSO = getResponse(requestNoSSO, method, url);
+        result.addAll(fromResponseReturnFoundProblems(responseNoSSO, endpoint, method, false));
+    }
+
+    private ResponseEntity<String> getResponse(HttpEntity<String> request, HttpMethod method, String url) {
+        ResponseEntity<String> response;
+        try {
+            response = restTemplate.exchange(url, method, request, String.class);
+        } catch (HttpClientErrorException | HttpServerErrorException e) {
+            response = ResponseEntity.status(e.getRawStatusCode())
+                .headers(e.getResponseHeaders())
+                .body(e.getResponseBodyAsString());
         }
+        return response;
+    }
+
+    private List<String> fromResponseReturnFoundProblems(ResponseEntity<String> response, Endpoint currentEndpoint, HttpMethod method, boolean attemptWithSSO) {
+        ArrayList<String> result = new ArrayList<>();
+
+        String responseBody = response.getBody();
 
+        if (responseBody != null && response.getStatusCode() == HttpStatus.NOT_FOUND && responseBody.contains("ZWEAM104E")) {
+            result.add("Documented endpoint at " + currentEndpoint.getUrl() + " could not be located, attempting to call it through gateway gives the ZWEAM104E error");
+        }
+
+        if (attemptWithSSO && responseBody != null && (response.getStatusCode() == HttpStatus.FORBIDDEN || response.getStatusCode() == HttpStatus.UNAUTHORIZED)) {
+            result.add(method + " request to documented endpoint at " + currentEndpoint.getUrl() + " responded with status code " + response.getStatusCode().value() + ", despite being called with the SSO authorization");
+        }
+
+        if (!currentEndpoint.isResponseCodeForMethodDocumented(String.valueOf(response.getStatusCode().value()), method)) {
+            result.add(method + " request to documented endpoint at " + currentEndpoint.getUrl() + " returns undocumented " + response.getStatusCode().value()
+                + " status code, documented responses are:" + currentEndpoint.getValidResponses().get(method.toString()));
+        }
         return result;
     }
 
-    public List<String> getProblemsWithEndpointUrls(AbstractSwaggerValidator swaggerParser) {
+    public static List<String> getProblemsWithEndpointUrls(AbstractSwaggerValidator swaggerParser) {
         return swaggerParser.getProblemsWithEndpointUrls();
     }
+
+
+    private String getAuthenticationCookie(String passedAuthenticationToken) {
+        if (passedAuthenticationToken.equals("dummy")) {
+            return tokenCreationService.createJwtTokenWithoutCredentials("validate");
```

Comment:
this should throw exception in case of zOSMF provider since for "validate" user it is not possible to generate passticket for zOSMF login.